lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -36,18 +36,22 @@ def start
       @zk = nil
 
       log.info "synapse: starting ZK watcher #{@name} @ hosts: #{@zk_hosts}, path: #{@discovery['path']}"
+      statsd.increment('synapse.zk_watcher.started')
       zk_connect
     end
 
     def stop
       log.warn "synapse: zookeeper watcher exiting"
+      statsd.increment('synapse.zk_watcher.clean_stop')
       zk_cleanup
     end
 
     def ping?
       # @zk being nil implies no session *or* a lost session, do not remove
       # the check on @zk being truthy
-      @zk && @zk.connected?
+      alive = @zk && @zk.connected?
+      statsd.increment("synapse.zk_watcher.ping.#{alive ? "success" : "failure"}")
```

Comment:
Moving this metric in the main loop would make it available for all watcher types.

Reply:
good point